src/cascadia/TerminalControl/ControlCore.h

```diff
@@ -239,6 +244,11 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         void _connectionOutputHandler(const hstring& hstr);
         void _updateHoveredCell(const std::optional<til::point> terminalPosition);
 
+        inline bool _IsClosing() const noexcept
+        {
+            return _closing.load();
+        }

```

Comment:
You can copy the variant from `TermControl` as is, allowing you to safely remove this incorrect use of an atomic.

Reply:
A suggestion:
```cpp
bool _closing{ false };

// ...

inline bool _IsClosing() const noexcept
{
#ifndef NDEBUG
	if (_dispatcher)
	{
	      // _closing isn't atomic and may only be accessed from the main thread.
	      assert(_dispatcher.HasThreadAccess());
	}
#endif
    return _closing;
}
```

Reply:
good thinking